tests/rptest/scale_tests/cloud_storage_compaction_test.py

```diff
@@ -0,0 +1,176 @@
+            "log_compaction_interval_ms",
+            str(self.configuration["log_compaction_interval_ms"]))
+        self._create_topic()
+        self.rr_cluster.start(start_si=False)
+        wait_until(self._create_read_repica_topic_success,
+                   timeout_sec=30,
+                   backoff_sec=5)
+
+    def _create_topic(self):
+        self.topic_spec = TopicSpec(
+            name=self.topic,
+            partition_count=self.configuration["partition_amount"],
+            replication_factor=3,
+            cleanup_policy="compact,delete",
+            segment_bytes=self.configuration["segment_size"],
+            retention_bytes=self.configuration["retention.local.target.bytes"])
+        self.kafka_tools.create_topic(self.topic_spec)
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic_spec.name, 'redpanda.remote.write',
+                               'true')
+
+    def _create_read_repica_topic_success(self):
+        try:
+            rpk_rr_cluster = RpkTool(self.rr_cluster)
+            conf = {
+                'redpanda.remote.readreplica':
+                self.si_settings.cloud_storage_bucket,
+            }
+            rpk_rr_cluster.create_topic(self.topic, config=conf)
+            return True
+        except RpkException as e:
+            if "The server experienced an unexpected error when processing the request" in str(
+                    e):
+                return False
+            else:
+                raise
+
+    def start_workload(self):
+        self.start_producer(
+            num_nodes=1,
+            throughput=self.configuration["throughput"],
+            repeating_keys=self.configuration["key_set_cardinality"])
+        wait_until(lambda: len(self.producer.last_acked_offsets) != 0, 30)
+
+    def tearDown(self):
+        assert self.redpanda and self.redpanda.s3_client
+        self.redpanda.s3_client.empty_bucket(self.s3_bucket_name)
+
+    @cluster(num_nodes=9)
+    def test_write(self):
+        self.start_workload()
+        self._rand_consumer = KgoVerifierRandomConsumer(
+            self.test_context, self.redpanda, self.topic, None,
+            self.configuration["random_message_validate"], 5)
+        self._rand_consumer.start()
+        self.start_consumer(num_nodes=1,
+                            redpanda_cluster=self.rr_cluster,
+                            verify_offsets=False)
+        while sum(self.producer.last_acked_offsets.copy().values()
+                  ) < self.configuration["produce_offsets"]:
+            last_acked_offsets = self.producer.last_acked_offsets.copy()
+            self.logger.error(f"Consuming until: {last_acked_offsets}")
+            self.await_consumed_offsets(last_acked_offsets, 240)
+            self.consumer.stop()
+            time.sleep(1)
+            self.consumer.start()
+        self.producer.stop()
+        self._rand_consumer.wait()
+        self.run_consumer_validation(enable_compaction=True)
```

Comment:
Please can you add checks on prometheus metrics here: the test should check that it really did some uploads/downloads, to avoid false-passes where the test accidentally just doesn't hit the cloud storage read path